src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtposition/WorkflowCompletionService.java

```diff
@@ -0,0 +1,110 @@
+package it.gov.pagopa.payhub.activities.service.ingestionflow.debtposition;
+
+import io.temporal.api.enums.v1.WorkflowExecutionStatus;
+import it.gov.pagopa.payhub.activities.connector.workflowhub.WorkflowHubService;
+import it.gov.pagopa.payhub.activities.dto.ingestion.debtposition.InstallmentErrorDTO;
+import it.gov.pagopa.payhub.activities.dto.ingestion.debtposition.InstallmentIngestionFlowFileDTO;
+import it.gov.pagopa.pu.workflowhub.dto.generated.WorkflowStatusDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Set;
+
+import static io.temporal.api.enums.v1.WorkflowExecutionStatus.*;
+
+@Service
+@Lazy
+@Slf4j
+public class WorkflowCompletionService {
+
+    private final WorkflowHubService workflowHubService;
+    private final int maxRetries;
+    private final int retryDelayMs;
+
+    private final Set<WorkflowExecutionStatus> TERMINAL_STATUSES = Set.of(
+            WORKFLOW_EXECUTION_STATUS_FAILED,
+            WORKFLOW_EXECUTION_STATUS_TERMINATED,
+            WORKFLOW_EXECUTION_STATUS_CANCELED,
+            WORKFLOW_EXECUTION_STATUS_TIMED_OUT,
+            WORKFLOW_EXECUTION_STATUS_COMPLETED
+    );
+
+    public WorkflowCompletionService(WorkflowHubService workflowHubService,
+                                     @Value("${ingestion-flow-files.dp-installments.wf-await.max-waiting-minutes:5}") double maxWaitingMinutes,
```

Comment:
int or long, why double?
```suggestion
                                     @Value("${ingestion-flow-files.dp-installments.wf-await.max-waiting-minutes:5}") int maxWaitingMinutes,
```

Reply:
I used double for maxWaitingMinutes because, with int, the formula `this.maxRetries = (int) ((maxWaitingMinutes * 60_000) / retryDelayMs);` would make maxWaitingMinutes round to 1 as the minimum value. This would lead to too many retries and longer test execution times.

I could switch to long maxWaitingMillis instead of int maxWaitingMinutes. This way, we could setvalues like 300ms for quick tests.

What do you think?

Reply:
you have to cast it during the calculation of the maxRetries, not expect it to be already a double!